shell/src/shell.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "shell/src/streams.h"
+#include "shell/src/parser.h"
+#include "shell/src/commands/command.h"
+
+#include <string>
+
+namespace shell {
+
+class Shell {
+public:
+    Shell(const std::shared_ptr<IStreams>& streams) : streams_(streams) {}
```

Comment:
I guess that IIState also needed here. Single global state can cause troubles in future (e.g. subshell must not change ENV after its execution)

Reply:
Do we really need to implement solutions for subshell? There is no such demands in technical requirement.

Reply:
I suggest not to implement, but to make it possible to do in the easier way. 
It requires only small changes, so why not?